spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -3112,9 +3112,9 @@ public void testAckModeCount() throws Exception {
 		container.start();
 		assertThat(commitLatch.await(10, TimeUnit.SECONDS)).isTrue();
 		assertThat(recordCount.get()).isEqualTo(7);
-		verify(consumer).commitSync(Collections.singletonMap(topicPartition, new OffsetAndMetadata(4L)),
+		verify(consumer).commitSync(Collections.singletonMap(topicPartition, new OffsetAndMetadata(3L)),
 				Duration.ofSeconds(42));
-		verify(consumer).commitSync(Collections.singletonMap(topicPartition, new OffsetAndMetadata(7L)),
+		verify(consumer).commitSync(Collections.singletonMap(topicPartition, new OffsetAndMetadata(6L)),
```

Comment:
Thanks for the contribution; in general it looks good but, in this case, the 7th record's offset will never be committed; I believe we need to accelerate the commit based on count if the next poll returns no records. This was not needed before this change.

Restoring the
```java
		verify(consumer).commitSync(Collections.singletonMap(topicPartition, new OffsetAndMetadata(7L)),
				Duration.ofSeconds(42));
```
(after the test for offset 6) causes the test to fail.

We also need to adjust the comments on lines 3063, 3068.

Reply:
Hi, @garyrussell 


**config**
AckMode is AckMode.COUNT
AckCount is 100

**expect action**
consumer.poll()  return 101 record    // commit offset 100th
consumer.poll()  return 0 record  // commit offset 101th
consumer.poll()  return 101 record  // commit offset 201th
consumer.poll()  return 0 record  // commit offset 202th
It looks like the submissions are a little frequent

**Change**
Can I change it here? Avoid commit offset when manually calling `Consumer.wake()`
https://github.com/spring-projects/spring-kafka/blob/ad5e754bf1c0b9c2566fe1ab900d33467ae07dd9/spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java#L1414-L1420